Mathlib.lean

```diff
@@ -3212,6 +3212,7 @@ import Mathlib.NumberTheory.Primorial
 import Mathlib.NumberTheory.PythagoreanTriples
 import Mathlib.NumberTheory.RamificationInertia
 import Mathlib.NumberTheory.Rayleigh
+import Mathlib.NumberTheory.SiegelsLemma
```

Comment:
I think `Siegel` is a better name for the file (judging on other names in mathlib), but this is not very important.

Reply:
The guy proved so many very important results that I felt it was not really fair to call `Siegel` a file with only this small (but fundamental) result.

I wonder if we should create a `DiophantineApproximation` folder inside `NumberTheory` and collect this file (which could then be called `Siegel` without shame), the files `DiophantineApproximation` and `WellApproximable` and the folder `Liouville` (I might be missing something) and put them in there.

Reply:
Yes, a folder is certainly a good idea (and you're right, a file called Siegel in `NumberTheory` is not very good...).

Reply:
Should I open a discussion on zulip?

Reply:
I think you can freely create a folder called DiophantineApproximation, but on the other hand I believe that the lemma should not be called Siegel's lemma (the file could, if needed). It would be better if the name "tells" what is going on.